hydra-node/src/Hydra/Party.hs

```diff
@@ -51,3 +58,26 @@ partyFromChain =
   maybe (fail "partyFromChain got Nothing") (pure . Party)
     . deserialiseFromRawBytes (AsVerificationKey AsHydraKey)
     . OnChain.partyToVerficationKeyBytes
+
+-- * Orphans
+
+instance ToJSON Plutus.PubKeyHash where
+  toJSON = \kh ->
+    object
+      [ "tag" .= Aeson.String "PubKeyHash"
+      , "keyHash" .= Aeson.String (decodeUtf8 $ Base16.encode $ fromBuiltin $ getPubKeyHash kh)
+      ]
+
+instance FromJSON Plutus.PubKeyHash where
+  parseJSON = withObject "PubKeyHash" $ \o -> do
+    tag <- o .: "tag"
+    case tag :: Text of
+      "PubKeyHash" -> do
+        hexText :: Text <- o .: "keyHash"
+        case Base16.decode $ encodeUtf8 hexText of
+          Left e -> fail e
+          Right bs -> pure $ Plutus.PubKeyHash (toBuiltin bs)
+      _ -> fail "Expected tag to be PubKeyHash"
+
+instance Arbitrary Plutus.PubKeyHash where
+  arbitrary = genericArbitrary
```

Comment:
What do you need these orphan instances for? Also, `Party` is a Hydra-concept and should NOT have anything to do with the `Plutus.PubKeyHash` / Cardano-specific L1 keys.

Must: remove these instances. No change to `Hydra.Party` should be necessary!

Reply:
I moved the json instance for `PubKeyHash` to the `Orphans` module in `hydra-plutus`. We need it since we use `[PubKeyHash]` in the `ClosedThreadOutput`.